x/ecocredit/server/basket/put_test.go

```diff
@@ -397,3 +401,267 @@ func assertUserSentCredits(t *testing.T, oldBalance math.Dec, amountSent string,
 
 	require.True(t, checkBalance.Equal(oldBalance))
 }
+
+type putSuite struct {
+	*baseSuite
+	blockTimestamp  time.Time
+	basketDenom     string
+	basketId        uint64
+	batch20110101   *ecocreditApi.BatchInfo
+	batch20110401   *ecocreditApi.BatchInfo
+	batch20110701   *ecocreditApi.BatchInfo
+	batch20120101   *ecocreditApi.BatchInfo
+	batch20120401   *ecocreditApi.BatchInfo
+	batch20120701   *ecocreditApi.BatchInfo
+	batch20130101   *ecocreditApi.BatchInfo
+	batch20130401   *ecocreditApi.BatchInfo
+	batch20130701   *ecocreditApi.BatchInfo
+	tradableCredits string
+	err             error
+}
+
+func TestPutDate(t *testing.T) {
+	gocuke.NewRunner(t, &putSuite{}).Path("../../features/basket/put_date.feature").Run()
+}
+
+func (s *putSuite) Before(t gocuke.TestingT) {
+	s.baseSuite = setupBase(t)
+	s.tradableCredits = "10"
+}
+
+func (s *putSuite) ACurrentBlockTimestampOf20220401() {
+	s.blockTimestamp = time.Date(2022, 04, 01, 0, 0, 0, 0, time.UTC)
+}
+
+func (s *putSuite) ABasketWithDateCriteriaYearsinthepastOf(a int64) {
+	var err error
+
+	s.basketDenom = "foo"
+
+	s.basketId, err = s.stateStore.BasketTable().InsertReturningID(s.ctx, &api.Basket{
+		BasketDenom:  s.basketDenom,
+		DateCriteria: &api.DateCriteria{YearsInThePast: 10},
+	})
+	assert.NilError(s.t, err)
+}
+
+func (s *putSuite) AUserOwnsCreditsFromABatchWithStartDate20110101() {
```

Comment:
You shouldn't need to define a method for each batch. Instead you should be able to setup the tests so that a single method takes some params to generate the batch. Try putting quotes around the dates in the feature file so this happens